src/Lecture2.hs

```diff
@@ -160,7 +224,60 @@ data Knight = Knight
     , knightEndurance :: Int
     }
 
-dragonFight = error "TODO"
+{- It's a sum data type, ok, primitive but works -}
+data Dragon a =
+  RedDragon { dragonGold :: Int, dragonFirepower :: Int, dragonHealth :: Int, dragonTreasure :: a }
+  | BlackDragon { dragonGold :: Int, dragonFirepower :: Int, dragonHealth :: Int, dragonTreasure :: a }
+  | GreenDragon { dragonGold :: Int, dragonFirepower :: Int, dragonHealth :: Int }
```

Comment:
I see a warning here about partial records. This happens because the `GreenDragon` constructor doesn't have the `dragonTreasure` field. This means that `dragonTreasure` function is partial 😞 

Usually, records in sum types are discouraged and plain products are used instead.

Otherwise, it's an amazing way of modelling this data type! 🔝 💯